text/src/autogluon/text/automm/optimization/lit_module.py

```diff
@@ -174,7 +178,12 @@ def training_step(self, batch, batch_idx):
         -------
         Average loss of the mini-batch data.
         """
-        output, loss = self._shared_step(batch)
+        if self.mixup_fn is not None and (self.hparams.mixup_off_epoch is None or self.current_epoch < self.hparams.mixup_off_epoch):
+            batch, target_a, target_b, lam = self.mixup_fn(batch, "timm_image_image", self.model.label_key)
+            output, _ = self._shared_step(batch)
+            loss = self._compute_loss(output=output, label=target_a) * lam + self._compute_loss(output=output,label=target_b) * (1. - lam)
```

Comment:
Why doesn't timm compute the loss in this way? https://github.com/rwightman/pytorch-image-models/blob/master/train.py#L703

Reply:
The mixup in timm can only be used in specific loss function. https://github.com/rwightman/pytorch-image-models/blob/d30685c283137b4b91ea43c4e595c964cd2cb6f0/train.py#L587-L592
The target in the output is the one-hot code of the mixed label which is related with the specific number of classes. https://github.com/rwightman/pytorch-image-models/blob/d30685c283137b4b91ea43c4e595c964cd2cb6f0/timm/data/mixup.py#L25-L27

Reply:
Can we also follow timm's practice to use its customized loss function?

Reply:
It is fine to use timm's loss functions in Multiclass and Binary. But it may face problems in Regression.

Reply:
We can define a similar loss class for regression. We choose the proper loss class based on the problem type. Using the modular design can make our training loop cleaner and scalable.

Reply:
There is no need to define a new loss class for regression. We only need to inherit timm's Mixup class and add a new target mixup util function for regression and a flag to choose which target mixup.